pkg/cache/cache.go

```diff
@@ -780,6 +777,19 @@ func (c *Cache) cleanupAssumedState(w *kueue.Workload) {
 	}
 }
 
+func (c *Cache) clusterQueueForWorkload(w *kueue.Workload) *ClusterQueue {
+	if w.Spec.Admission != nil {
+		return c.clusterQueues[string(w.Spec.Admission.ClusterQueue)]
+	}
+	wKey := workload.Key(w)
+	for _, cq := range c.clusterQueues {
+		if cq.Workloads[wKey] != nil {
+			return cq
+		}
+	}
```

Comment:
sigh... this loop is unfortunate. But we can get rid of this once we move the admission into status.

Reply:
That's right. Currently it is only used when cancelling admission.